core/common/src/main/java/alluxio/util/network/HttpUtils.java

```diff
@@ -30,32 +33,51 @@ private HttpUtils() {
   }
 
   /**
-   * use post method to send a url with arguments by http, this method can call RESTful Api.
+   * Uses the post method to send a url with arguments by http, this method can call RESTful Api.
+   *
    * @param url the http url
-   * @param timeout millisecond to wait for the server to respond before giving up
-   * @return the response body stream
-   * @throws IOException if an I/O error occurs
+   * @param timeout milliseconds to wait for the server to respond before giving up
+   * @param processInputStream the response body stream processor
+   * @return the response body stream if response status is OK or CREATED, otherwise return null
    */
-  public static InputStream post(String url, Integer timeout) throws IOException {
-    PostMethod postMethod = null;
+  public static String post(String url, Integer timeout,
+                                 IProcessInputStream processInputStream)
+      throws IOException {
+    Preconditions.checkNotNull(timeout, "timeout");
+    PostMethod postMethod = new PostMethod(url);
     try {
       HttpClient httpClient = new HttpClient();
-      if (null != timeout) {
-        httpClient.getHttpConnectionManager().getParams().setConnectionTimeout(timeout);
-        httpClient.getHttpConnectionManager().getParams().setSoTimeout(timeout);
-      }
-      postMethod = new PostMethod(url);
+      httpClient.getHttpConnectionManager().getParams().setConnectionTimeout(timeout);
+      httpClient.getHttpConnectionManager().getParams().setSoTimeout(timeout);
       int statusCode = httpClient.executeMethod(postMethod);
       if (statusCode == HttpStatus.SC_OK || statusCode == HttpStatus.SC_CREATED) {
-        return postMethod.getResponseBodyAsStream();
+        InputStream inputStream = postMethod.getResponseBodyAsStream();
+        if (processInputStream != null) {
+          processInputStream.process(inputStream);
+          return null;
+        } else {
+          StringBuilder contentBuffer = new StringBuilder();
+          try (BufferedReader br = new BufferedReader(
+              new InputStreamReader(inputStream, "UTF-8"))) {
+            String line;
+            while ((line = br.readLine()) != null) {
+              contentBuffer.append(line);
+            }
+          }
+          return contentBuffer.toString();
```

Comment:
@aaudiber i addressed @uronce-cc comments.

```
 uronce-cc on Jun 21 Contributor
instead of reading the whole body first and then parse it as json, a preferred way is to parse the response body as a json stream.
```

and i think it make sense, what the goal i return a string due to thinking about this is a util class, maybe other user want return string and do not care the input stream.